gobblin-runtime/src/main/java/org/apache/gobblin/runtime/job_catalog/NonObservingFSJobCatalog.java

```diff
@@ -105,6 +105,14 @@ public synchronized void remove(URI jobURI) {
       remove(jobURI, false);
   }
 
+  /**
+   * Removes a job from job catalog.
+   * If it is a flow delete request from service side, then alwaysTriggerListeners should be set to true so that a
+   * running job will also be cancelled. If it is a remove request because a job finishes or has been submitted to helix,
+   * then alwaysTriggerListeners should be set to false so that job cancellation is not triggered.
+   * @param jobURI job uri
+   * @param alwaysTriggerListeners if it is true, a running job will be cancelled.
```

Comment:
This architecture looks a bit strange to me. And looks like this API ambiguity was the source of the original bug.

Since removal from the catalog does not always mean cancelling the job, should this job cancellation be moved out of the catalog code and be done explicitly by the callers? E.g. caller can first cancel the job, if they want, and then remove it from the catalog.

Reply:
called the cancel on spec catalog listeners directly from job monitor

Reply:
passing EventBus object from JobConfigurationManager to JobMonitor and making JobMonitor directly posting a cancel event